File: Data Structures & Algorithms/partition-labels/submission-1.js
class Solution {
    /**
     * @param {string} S
     * @return {number[]}
     */
    partitionLabels(s) {
        let mpp = new Map()
        let n  = s.length 
        for(let i = n-1 ;i>=0 ;i--){
               if(!mpp.has(s[i])){
                mpp.set(s[i],i)
               }
        }

        let start = 0
        let end = 0
        let ans= []
        for(let i = 0;i < n ;i++ ){
            end = Math.max(end,mpp.get(s[i]))
            if(i==end){
                  ans.push(end-start+1)
                  start = i+1
            }
        }
        return ans 

    }
}

File: Data Structures & Algorithms/meeting-schedule-ii/submission-1.java
/**
 * Definition of Interval:
 * public class Interval {
 *     public int start, end;
 *     public Interval(int start, int end) {
 *         this.start = start;
 *         this.end = end;
 *     }
 * }
 */

class Solution {
    public int minMeetingRooms(List<Interval> intervals) {
  Collections.sort(intervals , (a,b)->{
    if(a.start!=b.start ) return  a.start -  b.start ;
    return a.end - b.end;

  } );

    
        PriorityQueue<Integer> pq = new PriorityQueue<>();

        int ans = 0;
        for (Interval meeting : intervals) {

            // A room has become free
            if (!pq.isEmpty() && pq.peek() <= meeting.start) {
                pq.poll();
            }

            // Allocate room for current meeting
            pq.offer(meeting.end);

            ans = Math.max(ans, pq.size());
        }
        return ans ;

    }
}
